pkg/auth/auth.go

```diff
@@ -239,24 +291,20 @@ func Logout(systemContext *types.SystemContext, opts *LogoutOptions, args []stri
 		return nil
 	}
 
-	err = config.RemoveAuthentication(systemContext, server)
+	err = config.RemoveAuthentication(systemContext, key)
 	switch errors.Cause(err) {
 	case nil:
-		fmt.Fprintf(opts.Stdout, "Removed login credentials for %s\n", server)
+		fmt.Fprintf(opts.Stdout, "Removed login credentials for %s\n", key)
 		return nil
 	case config.ErrNotLoggedIn:
-		authConfig, err := config.GetCredentials(systemContext, server)
```

Comment:
It’s not obvious to me that `GetCredentials` _before_ and _after` the `RemoveAuthentication` return the same values; AFAICS it’s possible for that not to be the case, if both `auth.json` and Docker configs contain credentials.

Reply:
Re-added this code path